lib/Zonemaster/Backend/DB.pm

```diff
@@ -130,6 +133,84 @@ sub process_unfinished_tests {
     }
 }
 
+sub _normalize_parameter_hash {
+    my ( $self, $params ) = @_;
+
+    my $profile = Zonemaster::Engine::Profile->effective;
+
+    my %normalized = ();
+    $normalized{domain}   = lc $$params{domain} // "";
+    $normalized{ipv4}     = $$params{ipv4}      // $profile->get( 'net.ipv4' );
+    $normalized{ipv6}     = $$params{ipv6}      // $profile->get( 'net.ipv6' );
+    $normalized{profile}  = $$params{profile}   // "default";
+    $normalized{priority} = $$params{priority}  // 10;
+    $normalized{queue}    = $$params{queue}     // 0;
+
+    my $array_ds_info = $$params{ds_info} // [];
+    my @array_ds_info_sort = sort {
+        (
+            defined $a->{algorithm} and
+            defined $b->{algorithm} and
```

Comment:
I added these checks everywhere because I got some warning on undefined values. Instead of looking for what should be defined or not when reaching this part, I decided to do the checks for all the keys indifferently.

Reply:
This has been removed as those values are required as described in the [API.md document](https://github.com/zonemaster/zonemaster-backend/blob/v6.2.0/docs/API.md#ds-info).